### PR TITLE
Partial support of the new Power-Board hardware

### PR DESCRIPTION
**Pull request by thomasfla**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description
# Firmware + SDK
This pull request introduces the support of the [power-board](https://github.com/open-dynamic-robot-initiative/power-board). With this board connected, the SDK will expose new sensor data: Bus voltage, Current consumption of the whole robot, energy drained by the robot since the power on. This is a first step toward using batteries on the ODRI robots.

Later feature will allow automatic shutdown of the robot triggered by over current and under voltage detection. The user will also be able to shut down the robot by software.

If used, the power board needs to be connected to the SPI7 connector.

The power board remain an optional hardware. This code will continue to work on hardware not connected to a powerboard.

### **Important note: The master board firmware AND sdk will need to be updated together, since this feature changes the communication protocol between the two.** 

# Hardware
Fix 0402 footprint to comply with seeed, fix HR911105A footprint, add labels for L1,U$3,U$4,U$5,U$6 ;  update board revision to V2.2
Fix BOM: correct C2 and R17 MPN; remove deprecated seeed BOM

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."

## How I Tested

Tested on ODRI hardware, Solo12 at LAAS for 2 months 

## Do not merge before

[//]: # "Link other open pull request here that need to be merged first."
[//]: # "Remove this section if it does not apply."

- [ ] link to dependent pull request

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [ ] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [ ] All new functions/classes are documented and existing documentation is updated according to changes.
- [ ] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: sdk/master_board_sdk/src/master_board_interface.cpp [hunk 1]
```diff
@@ -347,6 +347,12 @@ void MasterBoardInterface::ParseSensorData()
     imu_data.linear_acceleration[i] = D16QN_TO_FLOAT(sensor_packet.imu.linear_acceleration[i], IMU_QN_ACC);
   }
 
+  /*Read Power Board data*/
+  powerboard_data.current_bus = sensor_packet.powerboard.vshunt * 5e-6 / 4e-3; 
+  powerboard_data.voltage_bus = sensor_packet.powerboard.vbus * 3.125e-3;
+  powerboard_data.energy_bus = sensor_packet.powerboard.energy * 3.125e-3 * 5e-6 / 4e-3 ;
```
Comment (jviereck) [inline sdk/master_board_sdk/src/master_board_interface.cpp:353]: Two questions:
- should the constants used here be documented somewhere?
- small comment: I was not sure what `vshunt` stands for until reading this line and figuring out it means the electric current pulled by the motor board. Is `vshunt` the technical term or would it make sense to rename it to `cbus` or similar to make it more clear that it's about the current on the bus? This is only a small comment, so feel free to ignore it.

**Comment by jviereck** [comment]
@thomasfla , can this be merged?

**Comment by luator** [comment]
Yes, this branch is currently running on one of our Solo12 (which uses the powerboard).

**Comment by MaximilienNaveau** [comment]
@thomasfla @luator @nim65s I will merge this tomorrow evening if no complains.
Though there are quiet a bit of other PRs so O am not sure how much it will break the others work.

**Pull request merged**